app/src/main/java/com/asfoundation/wallet/backup/triggers/BackupTriggerPreferences.kt

```diff
@@ -0,0 +1,36 @@
+package com.asfoundation.wallet.backup.triggers
+
+import android.content.SharedPreferences
+import com.google.gson.Gson
+import javax.inject.Inject
+
+
+class BackupTriggerPreferences @Inject constructor(private val pref: SharedPreferences) {
```

Comment:
just an idea: access to sharedPrefs could also be treated as part of the data layer (like how it's made for the network or local DB), instead of directly accessing it on the fragment/viewModel. But I also agree with this way for the sharedPrefs